### PR TITLE
[Feature] Add logger warning to checkbox when name or value props have an empty value

### PR DESCRIPTION
**Pull request by LJKaski**
## Description
This PR adds a logger warning to checkbox when either name or value props have an empty value.

## Motivation and Context
Name and/or value props having empty values can lead to issues. Therefore it should be avoided. To help that, the library should warn the developer when an empty value for those props is given.

## How Has This Been Tested?
Tested by running locally and manually giving the props different values as well as with automated tests.

## Release notes
Added empty value or name warning to checkbox.

### CODE REVIEW

File: src/components/Form/Checkbox.tsx [hunk 1]
```diff
@@ -127,6 +127,15 @@ export class Checkbox extends Component<CheckboxProps> {
       );
     }
 
+    if (
+      ('name' in this.props && name === '') ||
```
Comment (aappoalander) [inline src/components/Form/Checkbox.tsx:131]: need to test for undef and null as well?

File: src/components/Form/Checkbox.tsx [hunk 2]
```diff
@@ -127,6 +127,15 @@ export class Checkbox extends Component<CheckboxProps> {
...
+      ('value' in this.props && value === '')
+    ) {
+      logger.warn(
+        'Name or value props should not have empty values. Provide valid values.',
```
Comment (aappoalander) [inline src/components/Form/Checkbox.tsx:135]: Maybe rephrase like "Name and value props should have non-empty values if provided."

File: src/components/Form/Checkbox.tsx (hunk 1 above)
Comment (LJKaski) [reply src/components/Form/Checkbox.tsx]: Good point. Should now cover those as well.

File: src/components/Form/Checkbox.tsx (hunk 2 above)
Comment (LJKaski) [reply src/components/Form/Checkbox.tsx]: Good suggestion. Now fixed.

**Review by aappoalander — APPROVED** [review]
